R/summarize_weighted_production.R

```diff
@@ -179,6 +179,19 @@ summarize_weighted_emission_factor <- function(data, ..., use_credit_limit = FAL
     ungroup()
 }
 
+summarize_unweighted_emission_factor <- function(data, ...) {
+  data <- data %>%
+    select(-c(
+      .data$id_loan,
+      .data$loan_size_credit_limit,
+      .data$loan_size_outstanding
+    )) %>%
+    distinct() %>%
+    group_by(.data$sector_ald, .data$year, ...) %>%
+    summarize(emission_factor_projected = mean(.data$emission_factor)) %>%
```

Comment:
nit: you can use `.groups = "drop"` in `summarise()` and skip the `ungroup()`

Reply:
Thank you, I actually prefer to call `ungroup()` explicitly (for totally aesthetic reasons).
Personally, I find it much easier to look through a dplyr chain for matching `group_by()` and `ungroup()` calls, than to look for a `.groups = "drop"` nested in a function call.

Reply:
fair enough, and agree it's easier to glance and pair up a `group_by()` with a `ungroup()`.... I kind of wish there were a `by` argument in `summarise()` so you could do something like...
```
data %>%
  summarise(emission_factor_projected = mean(.data$emission_factor), by = c(sector_ald, year))
```
instead of...
```
data %>%
  group_by(sector_ald, year, ...) %>%
  summarise(emission_factor_projected = mean(emission_factor)) %>%
  ungroup()
```